jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PushImageStep.java

```diff
@@ -107,7 +147,7 @@
       BuildContext buildContext,
       ProgressEventDispatcher.Factory progressEventDispatcherFactory,
       RegistryClient registryClient,
-      BuildableManifestTemplate manifestTemplate,
+      ManifestTemplate manifestTemplate,
       String imageQualifier,
       DescriptorDigest imageDigest,
       DescriptorDigest imageId) {
```

Comment:
I feel like this should be `@Nullable DescriptorDigest imageId` now that this doesn't apply to a manifest list.

That will trigger cascading changes to `BuildResult`, which requires a separate PR. I guess the `BuildResult.imageId` also needs to be `@Nullable` and `BuildResuilt.getImageId()` should return `Optional<DescriptorDigest>`.

Reply:
Hmm... now that we're pushing a manifest list, maybe it's better to make the class return a manifest digest only instead of a `BuildResult`.

Reply:
I can make a separate PR to do this change - which one do you prefer between the two.

Reply:
I think we need to discuss this with the team. We need to think about what to return in the end when multi-platform image building. This will most likely require changing the jib-core public API.

Reply:
Sounds Good .We can talk about this in the Jib Team Review tomorrow.